src/cachehash/main.py

```diff
@@ -122,18 +128,55 @@ def set(self, file_path: Union[str, Path], values: dict):
         if not file_path.exists():
             raise ValueError(f"{file_path} does not exist")
 
-        if isinstance(values, dict):
-            values = json.dumps(values, indent=4)
-        else:
-            raise ValueError("Must pass values as a dict")
+        values = self._values_to_str(values)
 
         hash = self.get_hash(file_path)
-        self.query(
-            "insert_record",
+        existing_record = self.query(
+            "get_record",
             {
                 "key": fp,
-                "hash": str(hash),
-                "value": values,
-            }
-        )
+            },
+        ).fetchone()
+        if existing_record is not None:
+            self.query(
+                "update_record",
+                {
+                    "key": fp,
+                    "hash": str(hash),
+                    "value": values,
+                },
+            )
+        else:
+            self.query(
+                "insert_record",
+                {
+                    "key": fp,
+                    "hash": str(hash),
+                    "value": values,
+                },
+            )
         self.db.commit()
+
+    def _values_to_str(self, values):
+        if isinstance(values, (dict, list)):
```

Comment:
what about tuple

Reply:
Good point. I'll try a tuple too

Reply:
still need to address tuple or we try writing it as string